From 8779efa7eef8af994bace64d4cc34765754606ac Mon Sep 17 00:00:00 2001
From: Chaitanya Godavarthi <chaitanya.kiran@netexperience.com>
Date: Sun, 2 May 2021 19:38:12 -0400
Subject: [PATCH] openwrt: eap101 sysupgrade on dual banks

Add sysupgrade code for flashing the image
on rootfs1 and rootfs2 and switch between
them.

Signed-off-by: Chaitanya Godavarthi <chaitanya.kiran@netexperience.com>
---
 .../ipq807x/base-files/lib/upgrade/platform.sh     | 14 +++++++++++++-
 target/linux/ipq807x/image/ipq60xx.mk              |  2 +-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/target/linux/ipq807x/base-files/lib/upgrade/platform.sh b/target/linux/ipq807x/base-files/lib/upgrade/platform.sh
index 59d1578925..82f01e8a99 100755
--- a/target/linux/ipq807x/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ipq807x/base-files/lib/upgrade/platform.sh
@@ -1,5 +1,8 @@
 . /lib/functions/system.sh
 
+RAMFS_COPY_BIN='fw_printenv fw_setenv'
+RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
+
 qca_do_upgrade() {
         local tar_file="$1"
 
@@ -57,7 +60,16 @@ platform_do_upgrade() {
 		nand_upgrade_tar "$1"
 		;;
 	edgecore,eap101)
-		CI_UBIPART="rootfs1"
+		part="$(awk -F 'ubi.mtd=' '{printf $2}' /proc/cmdline | sed -e 's/ .*$//')"
+		if [ "$part" = "rootfs1" ]; then
+			fw_setenv active 2 || exit 1
+			CI_UBIPART="rootfs2"
+		else
+			fw_setenv active 1 || exit 1
+			CI_UBIPART="rootfs1"
+		fi
+#		nand_do_upgrade "$1"
+#		CI_UBIPART="rootfs1"
 		nand_upgrade_tar "$1"
 		;;
 	esac
diff --git a/target/linux/ipq807x/image/ipq60xx.mk b/target/linux/ipq807x/image/ipq60xx.mk
index c536a174f8..af2ba13f30 100644
--- a/target/linux/ipq807x/image/ipq60xx.mk
+++ b/target/linux/ipq807x/image/ipq60xx.mk
@@ -25,7 +25,7 @@ define Device/edgecore_eap101
   DEVICE_DTS := qcom-ipq6018-edgecore-eap101
   DEVICE_DTS_CONFIG := config@cp01-c1
   SUPPORTED_DEVICES := edgecore,eap101
-  DEVICE_PACKAGES := ath11k-wifi-edgecore-eap101 uboot-env
+  DEVICE_PACKAGES := ath11k-wifi-edgecore-eap101 uboot-env uboot-envtools
 endef
 TARGET_DEVICES += edgecore_eap101
 
-- 
2.25.1

